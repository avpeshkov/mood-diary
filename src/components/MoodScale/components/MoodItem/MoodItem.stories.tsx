import React from "react";
import { action } from "@storybook/addon-actions";
import { withKnobs, number } from "@storybook/addon-knobs";
import { MoodItem } from "./MoodItem";
export default {
    title: "MoodItem",
    decorators: [withKnobs],
};

export const nonFilled = () => [
    <MoodItem key="1" onClick={action("MoodItem clicked")} mood={number("mood", 1)} isFilled={false} />,
    <MoodItem key="2" onClick={action("MoodItem clicked")} mood={number("mood2", 7)} isFilled={false} />,
];

export const Filled = () => [
    <MoodItem key="1" onClick={action("MoodItem clicked")} mood={number("mood", 1)} isFilled={true} />,
    <MoodItem key="2" onClick={action("MoodItem clicked")} mood={number("mood2", 6)} isFilled={true} />,
];
