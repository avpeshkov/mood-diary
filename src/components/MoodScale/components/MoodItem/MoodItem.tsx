import styled from "@emotion/styled";
import { css } from "@emotion/core";

const BaseCell = css`
    width: 25px;
    height: 25px;
    border: 1px solid;
    display: inline-block;
    border-radius: 15px;
    line-height: 25px;
    text-align: center;
    margin: 5px;
    vertical-align: bottom;
`;

const EmptyCell = css`
    border-color: black;
`;

const filledCell = (mood: Mood) => {
    return css`
        border-color: ${MoodColorMap[mood]};
        background-color: ${MoodColorMap[mood]};
        color: gray;
    `;
};

interface Props {
    isFilled: boolean;
    mood: Mood;
}

export const MoodItem = styled.button`
    ${BaseCell};
    ${({ isFilled, mood }: Props) => (isFilled ? filledCell(mood) : EmptyCell)};
`;

export const moods = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] as const;

export type Mood = typeof moods[number];

export const MoodColorMap: { [key in Mood]: string } = {
    1: "#372d21",
    2: "#606f4e",
    3: "#6978dd",
    4: "#5ecacc",
    5: "#a8d765",
    6: "#f8ea55",
    7: "#f8b94f",
    8: "#f46a3b",
    9: "#f86959",
    10: "#D60567",
};
