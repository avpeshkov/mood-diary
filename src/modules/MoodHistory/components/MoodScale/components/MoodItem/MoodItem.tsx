import styled from "@emotion/styled";
import { css } from "@emotion/core";
import { Mood } from "modules/MoodHistory/types";

const BaseCell = css`
    width: 25px;
    height: 25px;
    border: 1px solid;
    display: inline-block;
    border-radius: 15px;
    line-height: 25px;
    text-align: center;
    margin: 5px;
    vertical-align: bottom;
`;

const EmptyCell = css`
    border-color: black;
`;

const filledCell = (mood: Mood) => {
    return css`
        border-color: ${MoodColorMap[mood]};
        background-color: ${MoodColorMap[mood]};
        color: gray;
    `;
};

interface Props {
    isFilled: boolean;
    mood: Mood;
}

export const MoodItem = styled.button`
    ${BaseCell};
    ${({ isFilled, mood }: Props) => (isFilled ? filledCell(mood) : EmptyCell)};
`;

export const MoodColorMap: { [key in Mood]: string } = {
    1: "#17325a",
    2: "#7223b9",
    3: "#6978dd",
    4: "#5ecacc",
    5: "#a8d765",
    6: "#f8ea55",
    7: "#f8b94f",
    8: "#ea863f",
    9: "#f86959",
    10: "#e2200c",
};
