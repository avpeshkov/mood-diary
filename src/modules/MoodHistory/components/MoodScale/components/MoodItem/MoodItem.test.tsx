import React from "react";
import snapshotDiff from "snapshot-diff";

import { MoodItem } from "./MoodItem";

describe("MoodItem", () => {
    //  не уверен что это правильный способ прикрутить snapshot-diff, обясните пожалуйста если делаю что-то не так.
    test("no snapshot difference between equal components", () => {
        expect(
            snapshotDiff(<MoodItem onClick={jest.fn()} mood={1} isFilled={false} />, <MoodItem onClick={jest.fn()} mood={1} isFilled={false} />)
        ).toMatchSnapshot();
    });

    test("no snapshot difference between not filled components", () => {
        expect(
            snapshotDiff(<MoodItem onClick={jest.fn()} mood={1} isFilled={false} />, <MoodItem onClick={jest.fn()} mood={5} isFilled={false} />)
        ).toMatchSnapshot();
    });

    test(" no snapshot difference between  filled components", () => {
        expect(
            snapshotDiff(<MoodItem onClick={jest.fn()} mood={5} isFilled={true} />, <MoodItem onClick={jest.fn()} mood={5} isFilled={true} />)
        ).toMatchSnapshot();
    });
});
