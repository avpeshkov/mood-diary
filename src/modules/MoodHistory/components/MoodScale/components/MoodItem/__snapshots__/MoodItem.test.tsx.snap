// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodItem  no snapshot difference between  filled components 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`MoodItem no snapshot difference between equal components 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`MoodItem no snapshot difference between not filled components 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
